src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/UseEnvironmentMembersTests.cs

```diff
@@ -0,0 +1,390 @@
+
+        [Fact]
+        public async Task Diagnostics_ProcessPath_FixApplies_CSharp()
+        {
+            await VerifyCS.VerifyCodeFixAsync(
+@"
+using System;
+using System.Diagnostics;
+using System.Threading;
+
+namespace System
+{
+    public static class Environment
+    {
+        public static string ProcessPath => """";
+    }
+}
+
+class C
+{
+    string HandleProcessPath()
+    {
+        string path = {|CA1839:Process.GetCurrentProcess().MainModule.FileName|};
+        path = {|CA1839:Process.GetCurrentProcess().MainModule/*willberemoved*/.FileName|};
+        Use({|CA1839:Process.GetCurrentProcess().MainModule.FileName|});
+        Use(""test"",
+            {|CA1839:Process.GetCurrentProcess().MainModule.FileName|});
+        Use(""test"",
+            {|CA1839:Process.GetCurrentProcess().MainModule.FileName|} /* comment */,
+            0.0);
+        return {|CA1839:Process.GetCurrentProcess().MainModule.FileName|};
+    }
+
+    void Use(string path) {}
+    void Use(string something, string path) {}
+    void Use(string something, string path, double somethingElse) {}
+}
+",
+@"
+using System;
+using System.Diagnostics;
+using System.Threading;
+
+namespace System
+{
+    public static class Environment
+    {
+        public static string ProcessPath => """";
+    }
+}
+
+class C
+{
+    string HandleProcessPath()
+    {
+        string path = Environment.ProcessPath;
+        path = Environment.ProcessPath;
+        Use(Environment.ProcessPath);
+        Use(""test"",
+            Environment.ProcessPath);
+        Use(""test"",
+            Environment.ProcessPath /* comment */,
+            0.0);
+        return Environment.ProcessPath;
+    }
+
+    void Use(string path) {}
+    void Use(string something, string path) {}
+    void Use(string something, string path, double somethingElse) {}
+}
+");
+        }
+
+        [Fact]
+        public async Task Diagnostics_CurrentManagedThreadId_FixApplies_CSharp()
+        {
+            await VerifyCS.VerifyCodeFixAsync(
+@"
+using System;
+using System.Diagnostics;
+using System.Threading;
+
+namespace System
+{
+    public static class Environment
+    {
+        public static int CurrentManagedThreadId => 0;
+    }
+}
+
+class C
+{
+    int HandleCurrentManagedThreadId()
+    {
+        int id = {|CA1840:Thread.CurrentThread.ManagedThreadId|};
+        id = {|CA1840:Thread.CurrentThread/*willberemoved*/.ManagedThreadId|};
+        Use({|CA1840:Thread.CurrentThread.ManagedThreadId|});
+        Use(""test"",
+            {|CA1840:Thread.CurrentThread.ManagedThreadId|});
+        Use(""test"",
+            {|CA1840:Thread.CurrentThread.ManagedThreadId|} /* comment */,
+            0.0);
+        return {|CA1840:Thread.CurrentThread.ManagedThreadId|};
+    }
+
+    void Use(int id) {}
+    void Use(string something, int id) {}
+    void Use(string something, int id, double somethingElse) { }
+}
+",
+@"
+using System;
+using System.Diagnostics;
+using System.Threading;
```

Comment:
The codefix *could* cleanup the now-unused usings. This is not necessary but probably good to have as a follow-up. Should I open an issue tracking this?

Reply:
Do we do that with other analyzers? Removing usings seems orthogonal and a style preference,  not something we should be doing automatically just because we happened to remove the last use of something in that namespace. We could actually be creating more work and confusion for the developer.